transport/endpoint.go

```diff
@@ -1,12 +1,17 @@
 package transport
 
-import "github.com/aws/smithy-go"
+import (
+	"net/http"
+	"net/url"
+	"github.com/aws/smithy-go"
+)
 
-// Endpoint is a Smithy endpoint.
+// Endpoint is the endpoint object returned
+// by Endpoint resolution V2
 type Endpoint struct {
-	URI string
+	URI url.URL
 
-	Fields *FieldSet
+	Headers *http.Header
 
 	Properties smithy.Properties
```

Comment:
fix: add documentation to each member and how it's used, I know it feels self evident but we should document all public types/members.